server/traces/otel_http_converter.go

```diff
@@ -0,0 +1,112 @@
+package traces
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/kubeshop/tracetest/server/model"
+	v11 "go.opentelemetry.io/proto/otlp/common/v1"
+	v1 "go.opentelemetry.io/proto/otlp/trace/v1"
+)
+
+type HttpResourceSpans struct {
+	v1.ResourceSpans
+	ScopeSpans []*httpScopeSpans `json:"scopeSpans"`
+}
+
+type httpScopeSpans struct {
+	v1.ScopeSpans
+	Spans []*httpSpan `json:"spans"`
+}
+
+type httpSpan struct {
+	v1.Span
+	SpanId       string               `json:"spanId"`
+	ParentSpanId string               `json:"parentSpanId"`
+	Kind         string               `json:"kind"`
+	Attributes   []*httpSpanAttribute `json:"attributes"`
+}
+
+type httpSpanAttribute struct {
+	v11.KeyValue
+	Value map[string]interface{} `json:"value"`
+}
+
+func FromHttpOtelResourceSpans(resourceSpans []*HttpResourceSpans) model.Trace {
+	flattenSpans := make([]*httpSpan, 0)
+	for _, resource := range resourceSpans {
+		for _, scopeSpans := range resource.ScopeSpans {
+			flattenSpans = append(flattenSpans, scopeSpans.Spans...)
+		}
+	}
+
+	traceID := ""
+	spans := make([]model.Span, 0)
+	for _, span := range flattenSpans {
+		newSpan := convertHttpOtelSpanIntoSpan(span)
+		traceID = hex.EncodeToString(span.TraceId)
+		spans = append(spans, *newSpan)
+	}
+
+	return model.NewTrace(traceID, spans)
+}
+
+func convertHttpOtelSpanIntoSpan(span *httpSpan) *model.Span {
+	attributes := make(model.Attributes, 0)
+	for _, attribute := range span.Attributes {
+		attributes[attribute.Key] = getHttpAttributeValue(attribute.Value)
+	}
+
+	var startTime, endTime time.Time
+
+	if span.GetStartTimeUnixNano() != 0 {
+		startTime = time.Unix(0, int64(span.GetStartTimeUnixNano()))
+	}
+
+	if span.GetEndTimeUnixNano() != 0 {
+		endTime = time.Unix(0, int64(span.GetEndTimeUnixNano()))
+	}
+
+	spanID := createSpanID([]byte(span.SpanId))
+	attributes["parent_id"] = createSpanID([]byte(span.ParentSpanId)).String()
+
+	return &model.Span{
+		ID:         spanID,
+		Name:       span.Name,
+		StartTime:  startTime,
+		EndTime:    endTime,
+		Parent:     nil,
+		Children:   make([]*model.Span, 0),
+		Attributes: attributes,
+	}
+}
+
+func getHttpAttributeValue(value map[string]interface{}) string {
+	if value["stringValue"] != nil {
+		return value["stringValue"].(string)
+	}
+
+	if value["intValue"] != nil {
+		return fmt.Sprintf("%s", value["intValue"])
+	}
+
+	if value["doubleValue"] != nil {
+		val := value["doubleValue"].(float64)
+		if val != 0.0 {
```

Comment:
I'm not sure if this comparison will work correctly every time, because of the floating point craziness that we have (like `0.1 * 10 != 0.1 + 0.1 + 0.1 + 0.1 + 0.1 + 0.1 + 0.1 + 0.1 + 0.1 + 0.1` that we have in some languages like Python and JS).

To check if it is a double here we could do:
```go
val, ok := value["doubleValue"].(float64)
if ok {
  // do stuff of `if val != 0.0`
}
```